src/object.c

```diff
@@ -1842,3 +1845,126 @@ void memoryCommand(client *c) {
         addReplySubcommandSyntaxError(c);
     }
 }
+
+/*
+ * The parseExtendedStringArgumentsOrReply() function performs the common validation for extended
+ * string arguments used in SET and GET command.
+ *
+ * Get specific commands - PERSIST/DEL
+ * Set specific commands - XX/NX/GET/IFEQ
+ * Common commands - EX/EXAT/PX/PXAT/KEEPTTL
+ *
+ * Function takes pointers to client, flags, unit, pointer to pointer of expire obj if needed
+ * to be determined and command_type which can be COMMAND_GET or COMMAND_SET.
+ *
+ * If there are any syntax violations C_ERR is returned else C_OK is returned.
+ *
+ * Input flags are updated upon parsing the arguments. Unit and expire are updated if there are any
+ * EX/EXAT/PX/PXAT arguments. Unit is updated to millisecond if PX/PXAT is set.
+ */
+int parseExtendedStringArgumentsOrReply(client *c, int *flags, int *unit, robj **expire, robj **compare_val, int command_type, int max_args) {
+    int j = command_type == COMMAND_SET ? 3 : 2;
+    for (; j < max_args; j++) {
+        char *opt = c->argv[j]->ptr;
+        robj *next = (j == max_args - 1) ? NULL : c->argv[j + 1];
+
+        /* clang-format off */
+        if ((opt[0] == 'n' || opt[0] == 'N') &&
+            (opt[1] == 'x' || opt[1] == 'X') && opt[2] == '\0' &&
+            !(*flags & OBJ_SET_XX || *flags & OBJ_SET_IFEQ) && (command_type == COMMAND_SET))
+        {
+            *flags |= OBJ_SET_NX;
+        } else if ((opt[0] == 'x' || opt[0] == 'X') &&
+                   (opt[1] == 'x' || opt[1] == 'X') && opt[2] == '\0' &&
+                   !(*flags & OBJ_SET_NX || *flags & OBJ_SET_IFEQ) && (command_type == COMMAND_SET))
+        {
+            *flags |= OBJ_SET_XX;
+        } else if ((opt[0] == 'f' || opt[0] == 'F') &&
+            (opt[1] == 'n' || opt[1] == 'N') &&
+            (opt[2] == 'x' || opt[2] == 'X') && opt[3] == '\0' &&
+            !(*flags & OBJ_SET_FXX || *flags & OBJ_SET_IFEQ) && (command_type == COMMAND_HSET))
+        {
+            *flags |= OBJ_SET_FNX;
+        } else if ((opt[0] == 'f' || opt[0] == 'F') &&
+                   (opt[1] == 'x' || opt[1] == 'X') &&
+                   (opt[2] == 'x' || opt[2] == 'X') && opt[3] == '\0' &&
+                   !(*flags & OBJ_SET_FNX || *flags & OBJ_SET_IFEQ) && (command_type == COMMAND_HSET))
+        {
+            *flags |= OBJ_SET_FXX;
+        } else if ((opt[0] == 'i' || opt[0] == 'I') &&
+            (opt[1] == 'f' || opt[1] == 'F') &&
+            (opt[2] == 'e' || opt[2] == 'E') &&
+            (opt[3] == 'q' || opt[3] == 'Q') && opt[4] == '\0' &&
+            next && 
+            !(*flags & OBJ_SET_NX || *flags & OBJ_SET_XX || *flags & OBJ_SET_IFEQ) && (command_type == COMMAND_SET))
+        {
```

Comment:
This is within a `/* clang-format off */` block. Maybe it doesn't have to be? I don't know if clang-format would complain about anything here.

Anyway, these lines are misindented.

```suggestion
        } else if ((opt[0] == 'i' || opt[0] == 'I') &&
                   (opt[1] == 'f' || opt[1] == 'F') &&
                   (opt[2] == 'e' || opt[2] == 'E') &&
                   (opt[3] == 'q' || opt[3] == 'Q') && opt[4] == '\0' &&
                   next && 
                   !(*flags & OBJ_SET_NX || *flags & OBJ_SET_XX || *flags & OBJ_SET_IFEQ) &&
                   command_type == COMMAND_SET)
        {
```

Reply:
did not try, just took the code as it is. I can give it a try